specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,248 @@
+
+Ideally, if the app is installed per-machine then any activation registrations would also be per-machine; and if installed per-user then registrations would be per-user. The platform will not offer any additional API features for this - that is, the app doesn't get to choose whether its activation registrations are per-machine or per-user. Instead, the activations should be registered to match the way the app is installed. For v1, the platform will support only per-user registration.
+
+ <br>
+
+## Examples
+### Register for rich activation 
+Providing APIs for activation registration means that an app can register at any time. The most common case would be very early in the app?s main function. The app can register for any of the supported activation kinds. For most activation kinds, the app can register multiple options (for example, multiple filetypes or dataformats, or verbs) - and could therefore call the RegisterXXX function multiple times, if required. For example, an app could register for ".jpg" and ".bmp" filetype activations at one point, and then later also register for ".png". 
+
+Each registered filetype must be unique for the app. To allow for the case where the app determines the combination of options at runtime (and potentially changes the combinations at a later time), where an app registers the same filetype multiple times, the latest registration is honored and overwrites any previous registrations.
+
+For startup and toast activation, only one of each of these is allowed. If an app attempts to register either of these more than once, the latest registration is honored and overwrites any previous registrations.
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    // Registering for rich activation kinds can be done in the 
+    // app's installer or in the app itself.
+    RegisterForActivation();
+
+    // When the app starts, it can get its activated eventargs, and perform
+    // any required operations based on the activation kind and payload.
+    RespondToActivation();
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Standard Win32 window configuration/creation and message pump:
+    // ie, whatever the app would normally do - nothing new here.
+    RegisterClassAndStartMessagePump(hInstance, nCmdShow);
+    return 1;
+}
+
+void RegisterForActivation()
+{
+    // Get the AppLifecycle sub-object for activation registration.
+    ActivationRegistrationManager registrationManager = 
+        AppLifecycle::GetActivationRegistrationManager();
```

Comment:
Do we have any state that requires an instance class, or would this make more sense as a static class with register methods on it?

Reply:
I can imagine that at some point we might want a "get manager for sub application".  Doing that is usually a breaking change unless it's already an instance, but in the spirit of Project Reunion, feel free to avoid complexity and make these static methods.

Reply:
Updated.